src/aosm/azext_aosm/generate_nfd/cnf_nfd_generator.py

```diff
@@ -496,20 +506,27 @@ def traverse_dict(self, d, target):
         with the key being the deploy parameter and the value being the path to the value.
         e.g. {"foo": ["global", "foo", "bar"]}
 
-        param d: The dictionary to traverse.
-        param target: The regex to search for.
+        param d: The dictionary to traverse. param target: The regex to search for.
         """
         stack = [(d, [])]  # Initialize the stack with the dictionary and an empty path
         result = {}  # Initialize empty dictionary to store the results
         while stack:  # While there are still items in the stack
             # Pop the last item from the stack and unpack it into node (the dictionary) and path
             (node, path) = stack.pop()
-            for k, v in node.items():   # For each key-value pair in the popped item
+            for k, v in node.items():  # For each key-value pair in the popped item
                 if isinstance(v, dict):  # If the value is a dictionary
-                    stack.append((v, path + [k]))  # Add the dictionary to the stack with the path
-                elif isinstance(v, str) and re.search(target, v):  # If the value is a string + matches target regex
-                    match = re.search(target, v)  # Take the match i.e, foo from {deployParameter.foo}
-                    result[match.group(1)] = path + [k]  # Add it to the result dictionary with its path as the value
+                    stack.append(
+                        (v, path + [k])
+                    )  # Add the dictionary to the stack with the path
```

Comment:
I think having trailing comments is a bit weird here now.  Could you put them on separate lines?